ts/utils/biometrics.ts

```diff
@@ -2,52 +2,60 @@ import { Alert, Platform } from "react-native";
 import FingerprintScanner, {
   AuthenticateAndroid,
   AuthenticateIOS,
+  Biometrics,
   FingerprintScannerError
 } from "react-native-fingerprint-scanner";
 import { isDebugBiometricIdentificationEnabled } from "../config";
 import I18n from "../i18n";
 import { mixpanelTrack } from "../mixpanel";
-import { BiometrySimpleType } from "../sagas/startup/checkAcknowledgedFingerprintSaga";
+
+export type BiometricsErrorType =
+  // possibly working, but the string returned is undocumented
+  | "UNKNOWN"
+
+  // unspeakable horrors happened somewhere under the hood
+  | "UNAVAILABLE";
+
+export type BiometricsValidType =
+  // happy path
+  "BIOMETRICS" | "FACE_ID" | "TOUCH_ID";
+
+export type BiometricsType = BiometricsErrorType | BiometricsValidType;
 
 /**
  * Retrieve biometric settings from the base system. This function wraps the basic
  * method "isSensorAvailable" of react-native-fingerprint-scanner library and simplifies the possible returned values in
- * function of its usage:
- * * FaceID/TouchID/Biometrics: in case of biometric recognition supported.
- * * Throws an error in case the Biometric option is not enrolled or not available on the device
+ * function of its usage.
  *
  * More info about library can be found here: https://github.com/hieuvp/react-native-fingerprint-scanner
  */
-export function getFingerprintSettings(): Promise<BiometrySimpleType> {
-  return new Promise((resolve, _) => {
-    FingerprintScanner.isSensorAvailable()
-      .then(biometryType => {
-        switch (biometryType) {
-          case "Touch ID":
-            resolve("TOUCH_ID");
-            break;
-          case "Face ID":
-            resolve("FACE_ID");
-            break;
-          case "Biometrics":
-            resolve("BIOMETRICS");
-            break;
-          default:
-            resolve("UNAVAILABLE");
-            break;
-        }
-      })
-      .catch(e => {
-        void mixpanelTrack("BIOMETRIC_ERROR", { error: e });
-        resolve("UNAVAILABLE");
-      });
-  });
-}
+export const getBiometricsType = (): Promise<BiometricsType> =>
+  FingerprintScanner.isSensorAvailable()
+    .then((biometryType: Biometrics) => {
+      switch (biometryType) {
+        case "Touch ID":
+          return "TOUCH_ID";
+        case "Face ID":
+          return "FACE_ID";
+        case "Biometrics":
+          return "BIOMETRICS";
+        default:
+          return "UNKNOWN";
+      }
+    })
+    .catch(e => {
+      void mixpanelTrack("BIOMETRIC_ERROR", { error: e });
```

Comment:
```suggestion
      void mixpanelTrack("BIOMETRIC_ERROR", { reason: e.message });
```
Since we track all failure using the prop `reason`, could it be a good idea to pass only the message and not an object?